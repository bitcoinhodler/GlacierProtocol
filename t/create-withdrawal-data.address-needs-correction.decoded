{
  "txid": "45abeb47533e211f7c314c265309d80e7a593978dcd9ae71b8910e7d55e8a7e0",
  "hash": "45abeb47533e211f7c314c265309d80e7a593978dcd9ae71b8910e7d55e8a7e0",
  "version": 2,
  "size": 402,
  "vsize": 402,
  "weight": 1608,
  "locktime": 0,
  "vin": [
    {
      "txid": "91e963fa4ae5a194e8dfc936e6e5bdfb3aaf166d8e666900f44fb9a74069fd91",
      "vout": 1,
      "scriptSig": {
        "asm": "0 304402207017e0e4d4c26cc7ffe928ca51ee79f0f8d7d2de5fbc008847f9a400c0a27df9022031a84af21d92e8d9e28d70f86f15c964922c4ddb29fad0b5fa3b7e8fd84b302b[ALL] 304402200091b634fc996d9fb5304cacfb48acd9a13183f2a2b237d158fccf8a4a54f6aa02204b6c0c7cd2a992bd856170b393263b7c097028e18296064ed06348ed6ad76946[ALL] 52210277f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece178113210203f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac21026895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c621039f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff654ae",
        "hex": "0047304402207017e0e4d4c26cc7ffe928ca51ee79f0f8d7d2de5fbc008847f9a400c0a27df9022031a84af21d92e8d9e28d70f86f15c964922c4ddb29fad0b5fa3b7e8fd84b302b0147304402200091b634fc996d9fb5304cacfb48acd9a13183f2a2b237d158fccf8a4a54f6aa02204b6c0c7cd2a992bd856170b393263b7c097028e18296064ed06348ed6ad76946014c8b52210277f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece178113210203f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac21026895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c621039f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff654ae"
      },
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.00000580,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 606fb72624603795d73b911a729d88f95f9b6ac8 OP_EQUAL",
        "desc": "addr(2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL)#ly84wms3",
        "hex": "a914606fb72624603795d73b911a729d88f95f9b6ac887",
        "address": "2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL",
        "type": "scripthash"
      }
    },
    {
      "value": 0.00000400,
      "n": 1,
      "scriptPubKey": {
        "asm": "0 6e88bd9437ce56f7800059a8d4d137f557880ccd",
        "desc": "addr(bcrt1qd6ytm9pheet00qqqtx5df5fh74tcsrxdm7qpqa)#ary6unm3",
        "hex": "00146e88bd9437ce56f7800059a8d4d137f557880ccd",
        "address": "bcrt1qd6ytm9pheet00qqqtx5df5fh74tcsrxdm7qpqa",
        "type": "witness_v0_keyhash"
      }
    }
  ]
}
