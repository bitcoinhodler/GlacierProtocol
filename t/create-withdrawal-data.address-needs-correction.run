#!/bin/bash
set -e

# Enter No on the final confirmation, starting over the withdrawal
# process. Enter a new destination address the second time through.

$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT 2>&1
y
y
y
y
y
y
2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
52210277f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece178113210203f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac21026895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c621039f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff654ae
mp1KQzUoDf9div7jjgon3UuqLThMSo3tmF
1
0200000001a7a5665319a117ec78387fc59ed0c20714c9143bd8093452b9ed55940f00be57010000006a47304402204ceb2688c3d7ada2dbaa56de6916b51575eea9216f89b9a2c342560ab91e919f022041bad6a0d601c7ab3faf91ea9fe1e12e08d7d896b3046998d9f939d41dfef7aa0121035766f287bb612c0a29219d44d6d0c0987c5b1e7931f4ccde37839152256d76daffffffff023d1ffd00000000001976a914c1a7280bd89be84b7c3ba07ed02c72448f28d1c888ac881300000000000017a914606fb72624603795d73b911a729d88f95f9b6ac88700000000
2
cSfvoUsWWZ81M4qGYciPHdNxHXQxPEsXBdGWjYdchpisPabtiKVQ
cPWVKeCBjnv833Kayc6nczBbG4YqqbCucup3n5D68WPzBEfceBAQ
1
n
10
y
0.00000400
q
n
2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
52210277f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece178113210203f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac21026895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c621039f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff654ae
bcrt1qd6ytm9pheet00qqqtx5df5fh74tcsrxdm7qpqa
1
0200000001a7a5665319a117ec78387fc59ed0c20714c9143bd8093452b9ed55940f00be57010000006a47304402204ceb2688c3d7ada2dbaa56de6916b51575eea9216f89b9a2c342560ab91e919f022041bad6a0d601c7ab3faf91ea9fe1e12e08d7d896b3046998d9f939d41dfef7aa0121035766f287bb612c0a29219d44d6d0c0987c5b1e7931f4ccde37839152256d76daffffffff023d1ffd00000000001976a914c1a7280bd89be84b7c3ba07ed02c72448f28d1c888ac881300000000000017a914606fb72624603795d73b911a729d88f95f9b6ac88700000000
2
cSfvoUsWWZ81M4qGYciPHdNxHXQxPEsXBdGWjYdchpisPabtiKVQ
cPWVKeCBjnv833Kayc6nczBbG4YqqbCucup3n5D68WPzBEfceBAQ
10
y
0.00000400
y
INPUT
