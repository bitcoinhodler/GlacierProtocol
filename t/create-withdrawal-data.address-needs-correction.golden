Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.00005000 BTC

How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00000404 bitcoin.
Confirm? (y/n): 
Fee calculation aborted. Starting over...

Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00004040 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00004040, you have 0.00000960 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to mp1KQzUoDf9div7jjgon3UuqLThMSo3tmF (leave blank to withdraw all funds stored in these unspent transactions): 0.00000560 being returned to cold storage address address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.00005000 BTC in unspent inputs from cold storage address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
0.00000560 BTC going back to cold storage address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
0.00000400 BTC going to destination address mp1KQzUoDf9div7jjgon3UuqLThMSo3tmF
Fee amount: 0.00004040

Signing with private keys: 
cSfvoUsWWZ81M4qGYciPHdNxHXQxPEsXBdGWjYdchpisPabtiKVQ
cPWVKeCBjnv833Kayc6nczBbG4YqqbCucup3n5D68WPzBEfceBAQ


Confirm? (y/n): You must enter y (for yes) or n (for no).
Confirm? (y/n): 
Process aborted. Starting over....

You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.00005000 BTC

How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00004020 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00004020, you have 0.00000980 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to bcrt1qd6ytm9pheet00qqqtx5df5fh74tcsrxdm7qpqa (leave blank to withdraw all funds stored in these unspent transactions): 0.00000580 being returned to cold storage address address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.00005000 BTC in unspent inputs from cold storage address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
0.00000580 BTC going back to cold storage address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
0.00000400 BTC going to destination address bcrt1qd6ytm9pheet00qqqtx5df5fh74tcsrxdm7qpqa
Fee amount: 0.00004020

Signing with private keys: 
cSfvoUsWWZ81M4qGYciPHdNxHXQxPEsXBdGWjYdchpisPabtiKVQ
cPWVKeCBjnv833Kayc6nczBbG4YqqbCucup3n5D68WPzBEfceBAQ


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 10.00 satoshis per vbyte

Raw signed transaction (hex):
020000000191fd6940a7b94ff40069668e6d16af3afbbde5e636c9dfe894a1e54afa63e99101000000fd1e010047304402207017e0e4d4c26cc7ffe928ca51ee79f0f8d7d2de5fbc008847f9a400c0a27df9022031a84af21d92e8d9e28d70f86f15c964922c4ddb29fad0b5fa3b7e8fd84b302b0147304402200091b634fc996d9fb5304cacfb48acd9a13183f2a2b237d158fccf8a4a54f6aa02204b6c0c7cd2a992bd856170b393263b7c097028e18296064ed06348ed6ad76946014c8b52210277f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece178113210203f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac21026895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c621039f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff654aeffffffff02440200000000000017a914606fb72624603795d73b911a729d88f95f9b6ac88790010000000000001600146e88bd9437ce56f7800059a8d4d137f557880ccd00000000

Transaction fingerprint (md5):
eb79314ad1d87e8e575e8a8694bc6799
QR code for transaction written to transaction.png
